backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -365,4 +418,362 @@ public FutureExperimentRunDAO(Executor executor, FutureJdbi jdbi, UAC uac) {
         .thenCompose(unused -> artifactHandler.deleteArtifacts(runId, optionalKeys), executor)
         .thenCompose(unused -> updateModifiedTimestamp(runId, now), executor);
   }
+
+  private InternalFuture<UserInfo> getCurrentLoginUserInfo() {
+    return FutureGrpc.ClientRequest(
+        uac.getUACService().getCurrentUser(Empty.newBuilder().build()), executor);
+  }
+
+  public InternalFuture<ExperimentRun> createExperimentRun(final CreateExperimentRun request) {
+    // Validate arguments
+    var futureTask =
+        InternalFuture.runAsync(
+            () -> {
+              String errorMessage = null;
+              if (request.getProjectId().isEmpty() && request.getExperimentId().isEmpty()) {
```

Comment:
No need to check both, then check one and the other. This leads to exponential growth in checks. Just check the individual fields.

Reply:
Removed this check